languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -20273,7 +20273,66 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
             </rule>
         </rulegroup>
 
-
+        <!-- My assumption here is that these are mostly typos since they're pretty obvious ungrammatical,
+             from the descriptive point of view. So it doesn't seem to be worthwhile to account for non-3SG
+             inflections. I mean, 'dever' -> 'deve' is pretty straightforward, but 'dever' -> 'devo' less so.  -->
+        <rulegroup id="DOUBLE_INFINITIVE" name="Infinitivo duplo">
+            <short>Possível erro de digitação.</short>
+            <!-- This contains the original structure as described in premium issue 5538
+                 All sentences herein are of the shape: 'dever ter feito', and as such will usually indicate
+                 some kind of past tense epistemic modality, for which the double infinitive is very probably ungrammatical. -->
+            <rule default="temp_off">
+                <!-- This antipattern refers to constructions of the type 'the fact that...', for which
+                     a personal (i.e. inflected) infinitive can be used. -->
+                <antipattern>
+                    <token regexp="yes">fac?to|questão|possibilidade|hipótese|apesar</token>
+                    <token regexp="yes" skip="-1" postag_regexp="yes" postag="SPS00(:.+)?">
+                        d([eoa]|es[st][ea]|aquel[ea]|uma?|is[st]o)
```

Comment:
We can add more instances: `d([eoa]s?|es[st][ea]s?|aquel[ea]?|aquilo?|uma?|uns|um|is[st]o)`. This could also become an entity, as there are more rules which might require limiting the `SPS*` tag to this.

Reply:
Checking the pattern now, this rule does not trigger `d([eoa]s?|es[st][ea]s?|aquel[ea]?|aquilo?|uma?|uns|um|is[st]o)` anyway because the first token in the pattern is limited to `SENT_START|[DNPARZ].+`. So are the antipatterns necessary?

Reply:
Not sure what you mean. The first token of the pattern includes `N.+` postags, which should match the words in the first antipattern token. Or am I seeing things? Seemed to work as expected as I was testing it.

Reply:
Nope, at least here `N.+` doesn't show up, only` SPS*` and `V*`:
<img width="331" alt="image" src="https://github.com/languagetool-org/languagetool/assets/108675843/2e8a83c4-2040-4ec2-bb14-8de2d8eab3e1">

Reply:
![anigif_sub-buzz-15209-1490296003-28](https://github.com/languagetool-org/languagetool/assets/50704700/8bc04c6d-2be0-4bb5-a4ee-dd7da5f10c1e)

Reply:
Just realized I missed the `skip="-1"` in the antipattern. 

Just adding `d([eoa]s?|es[st][ea]s?|aquel[ea]?|aquilo?|uma?|uns|um|is[st]o)` should do the deed 😄.

Reply:
I really don't see what wouldn't be working. The tests pass, and when I change it they don't.

Reply:
Just realized I completely missed `skip="-1"` in the antipattern. Let's just add `d([eoa]s?|es[st][ea]s?|aquel[ea]?|aquilo?|uma?|uns|um|is[st]o)`.

Reply:
And also `um` and the plural form of the others (this is what we have right now:` d([eoa]|es[st][ea]|aquel[ea]|uma?|is[st]o`)

Reply:
Oh, right, now I remembered. I had originally chosen not to accept the plural forms precisely because I assumed that `dever` was a misspelling for `deve`. Adding plural also tacks on the extra complication of there being a further number agreement error.

Reply:
True, let's leave it as it is and wait for the results 😄